addons/pos_adyen/models/adyen_account.py

```diff
@@ -0,0 +1,115 @@
+        }
+
+
+class AdyenStore(models.Model):
+    _name = 'adyen.store'
+    _inherit = ['adyen.address.mixin']
+    _description = 'Adyen Store'
+
+    adyen_account_id = fields.Many2one('adyen.account', ondelete='cascade')
+    store_uuid = fields.Char('UUID', readonly=True)
+    store_reference = fields.Char('Reference', default=lambda self: uuid.uuid4().hex)
+    name = fields.Char('Name', required=True)
+    phone_number = fields.Char('Phone Number', required=True)
+    terminal_ids = fields.One2many('adyen.terminal', 'store_id', string='Payment Terminals', readonly=True)
+
+    @api.model
+    def create(self, values):
+        adyen_store_id = super(AdyenStore, self).create(values)
+        response = adyen_store_id.adyen_account_id._adyen_rpc('update_account_holder', adyen_store_id._format_data())
+        stores = response['accountHolderDetails']['storeDetails']
+        created_store = next(store for store in stores if store['storeReference'] == adyen_store_id.store_reference)
+        adyen_store_id.with_context(update_from_adyen=True).sudo().write({
+            'store_uuid': created_store['store'],
+        })
+        return adyen_store_id
+
+    def unlink(self):
+        for store_id in self:
+            store_id.adyen_account_id._adyen_rpc('close_stores', {
+                'accountHolderCode': store_id.adyen_account_id.account_holder_code,
+                'stores': [store_id.store_uuid],
+            })
+        return super(AdyenStore, self).unlink()
+
+    def _format_data(self):
+        return {
+            'accountHolderCode': self.adyen_account_id.account_holder_code,
+            'accountHolderDetails': {
+                'storeDetails': [{
+                    'storeReference': self.store_reference,
+                    'storeName': self.name,
+                    'merchantAccount': 'OdooMP_POS',
+                    'merchantCategoryCode': '7999',
+                    'address': {
+                        'city': self.city,
+                        'country': self.country_id.code,
+                        'houseNumberOrName': self.house_number_or_name,
+                        'postalCode': self.zip,
+                        'stateOrProvince': self.state_id.code or None,
+                        'street': self.street,
+                    },
+                    'fullPhoneNumber': self.phone_number,
+                }],
+            }
+        }
+
+
+class AdyenTerminal(models.Model):
+    _name = 'adyen.terminal'
+    _description = 'Adyen Terminal'
+    _rec_name = 'terminal_uuid'
+
+    adyen_account_id = fields.Many2one('adyen.account', ondelete='cascade')
+    store_id = fields.Many2one('adyen.store')
+    terminal_uuid = fields.Char('Terminal ID')
+
+    @api.model
+    def _sync_adyen_terminals(self):
+        for adyen_store_id in self.env['adyen.store'].search([]):
+            response = adyen_store_id.adyen_account_id._adyen_rpc('connected_terminals', {
+                'merchantAccount': 'OdooMP_POS',
```

Comment:
I haven't checked the proxy yet, but could it be possible to move this in the proxy "'merchantAccount': 'OdooMP_POS',"
Or if it is not possible maybe an ir_config_parameter